flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/schema/factories/KeyValueSchemaFactory.java

```diff
@@ -69,11 +69,8 @@ public SchemaType type() {
     public TypeInformation<KeyValue<K, V>> createTypeInfo(SchemaInfo info) {
         KeyValue<SchemaInfo, SchemaInfo> kvSchemaInfo = decodeKeyValueSchemaInfo(info);
 
-        Schema<K> keySchema = PulsarSchemaUtils.createSchema(kvSchemaInfo.getKey());
```

Comment:
TODO: this is something need to consult yufan. Why did we first create the schema and then get the schema info ?

Reply:
This change is because for STRING types, after the create-create operation the type class-info will be lost~ Good to know it's ok to skip the validation ~